### DISCUSSION TITLE
Add multiple edges between the same pair of nodes, each with different attributes and specified colors

### DISCUSSION THREAD

**Discussion by XM-8JD2** [cosmograph-org/cosmos#111] [Ideas]
Currently, edges do not have attributes, which prevents important information from being displayed during graph analysis (for example, one user following another user or one user reporting another user). Storing and visualizing edge attributes should be independent of the existing vertex force feedback implementation, as this is purely a UI issue.

Proposed Implementation:

Display the edge's label near the vertex when the mouse hovers over the edge.

When there are multiple edges from the same vertex to the same target vertex in the same direction, use different colors to distinguish them.
Consider rendering edges as "colorful dashed lines," with each edge in that direction assigned a different color.

Render bidirectional edges as two separate lines between the vertices, conveying all necessary information clearly.

![khh](https://github.com/cosmograph-org/cosmos/assets/133830244/2a81614a-9586-4902-adb7-3a67f65c147e)

**Comment by rokotyan** [comment]
Thanks for the feedback @XM-8JD2! We'll consider this, but rendering multiple links and rendering link labels are currently on our priority list.

**Reply by XM-8JD2** [reply]
What you said is right, that is more important
